apis/gateway/v1beta1/apirule_types.go

```diff
@@ -85,6 +85,7 @@ func (s *APIRuleStatus) ApiRuleStatusVersion() versions.Version {
 // +kubebuilder:subresource:status
 // +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.APIRuleStatus.code"
 // +kubebuilder:printcolumn:name="Host",type="string",JSONPath=".spec.host"
+// +kubebuilder:deprecatedversion:warning="This version of APIRule is deprecated and will be removed in future releases. Please use v2 instead."
```

Comment:
Those deprecation notes should include which version they are pointing to, as generally it might not be visible by the user during apply, and instead e.g. when checking for general api deprecations on cluster
```suggestion
// +kubebuilder:deprecatedversion:warning="v1beta1 version of APIRule is deprecated and will be removed in future releases. Please use v2 instead."
```